pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -133,6 +135,17 @@ func (cl *Client) HasSynced() bool {
 	return true
 }
 
+func (cl *Client) SyncErr() error {
+	req, _ := klabels.NewRequirement("noop.istio.io/noop", "=", []string{"noop"})
+	var errs error
+	for _, ctl := range cl.kinds {
+		if _, err := ctl.lister("default").List(klabels.NewSelector().Add(*req)); err != nil {
```

Comment:
the errors we see in `klog` happen somewhere inside the informers where we can't react or detect them..  resorting to calling listers unless we have a better idea

Reply:
After a closer look at List method, i think it is not right to judge the access to remote clusters based on the error returned. In reality, there is no error even the client can not access remote cluster.

Reply:
@stevenctl note this